src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs

```diff
@@ -452,13 +389,13 @@ bool ITaskExecutionHost.SetTaskParameters(IDictionary<string, Tuple<string, Elem
         /// <returns>True of the outputs were gathered successfully, false otherwise.</returns>
         bool ITaskExecutionHost.GatherTaskOutputs(string parameterName, ElementLocation parameterLocation, bool outputTargetIsItem, string outputTargetName)
         {
-            ErrorUtilities.VerifyThrow(_taskFactoryWrapper != null, "Need a taskFactoryWrapper to retrieve outputs from.");
+            ErrorUtilities.VerifyThrow(_UNITTESTONLY_TaskFactoryWrapper != null, "Need a taskFactoryWrapper to retrieve outputs from.");
```

Comment:
Going to an autoprop makes this look very wrong here. It's _used_ everywhere, but should be _set_ only in unit tests. I think I'd rather see this reverted.

Reply:
You're right, I thought I reverted this already. Fixed.